File: src/main/kotlin/com/projecthub/events/project/ProjectCreatedEvent.java
package com.projecthub.core.domain.event.project;

import com.projecthub.core.domain.event.DomainEvent;
import com.projecthub.core.domain.model.project.ProjectId;

public class ProjectCreatedEvent extends DomainEvent {
    private final ProjectId projectId;
    private final String name;

    public ProjectCreatedEvent(ProjectId projectId, String name) {
        super();
        this.projectId = projectId;
        this.name = name;
    }

    public ProjectId getProjectId() {
        return projectId;
    }

    public String getName() {
        return name;
    }
}

File: src/main/kotlin/com/projecthub/domain/project/events/ProjectCreatedEvent.kt
package com.projecthub.domain.project.events

import com.projecthub.events.DomainEvent
import java.time.Instant
import java.util.UUID

/**
 * Event representing project creation.
 */
class ProjectCreatedEvent(
    override val aggregateId: UUID,
    val name: String,
    val description: String,
    val startDate: Instant,
    val dueDate: Instant? = null,
    val ownerId: UUID
) : DomainEvent() {
    override val type: String = "ProjectCreatedEvent"
}
